From 54a61cd7b5e6c2cbba78556bb81eb6dcbe11c47c Mon Sep 17 00:00:00 2001
From: Daniel Goehring <dgoehrin@os.amperecomputing.com>
Date: Thu, 4 Mar 2021 12:32:54 -0500
Subject: [PATCH 24/31] QS|MQ: BMC remote debug user configs [DO NOT UPSTREAM]

Add AST2500 BMC remote debug user configuration files for Ampere Altra
("Quicksilver") and Ampere Altra Max ("Mystique").

Tested on Altra and Altra Max silicon systems with AST2500 BMC

Change-Id: I45c93228e0fdd6bb8f5f91eeee5e6700ad0bdfa2
Signed-off-by: Daniel Goehring <dgoehrin@os.amperecomputing.com>
---
 tcl/openocd.mq_1s_bmc.cfg | 87 +++++++++++++++++++++++++++++++++++
 tcl/openocd.mq_2s_bmc.cfg | 95 +++++++++++++++++++++++++++++++++++++++
 tcl/openocd.qs_1s_bmc.cfg | 86 +++++++++++++++++++++++++++++++++++
 tcl/openocd.qs_2s_bmc.cfg | 94 ++++++++++++++++++++++++++++++++++++++
 4 files changed, 362 insertions(+)
 create mode 100644 tcl/openocd.mq_1s_bmc.cfg
 create mode 100644 tcl/openocd.mq_2s_bmc.cfg
 create mode 100644 tcl/openocd.qs_1s_bmc.cfg
 create mode 100644 tcl/openocd.qs_2s_bmc.cfg

diff --git a/tcl/openocd.mq_1s_bmc.cfg b/tcl/openocd.mq_1s_bmc.cfg
new file mode 100644
index 000000000..6f4739b1a
--- /dev/null
+++ b/tcl/openocd.mq_1s_bmc.cfg
@@ -0,0 +1,87 @@
+#
+# Example OpenOCD user configuration for the AST2500 BMC
+# on the Ampere development platforms
+#
+# Copyright (c) 2020-2021, Ampere Computing LLC
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of
+# the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program;
+#
+#
+
+# Command Line Argument Description
+#
+# JTAGFREQ
+# Override the default JTAG clock frequency
+# If not specified, defaults to "12375" kHz
+# Syntax: -c "set JTAGFREQ {12375}"
+#
+# SYSNAME
+# Override the default system name
+# If not specified, defaults to "mq"
+# Syntax: -c "set SYSNAME {mq}"
+#
+# Life-Cycle State (LCS)
+# If not specified, defaults to "Secure LCS"
+# LCS=0, "Secure LCS"
+# LCS=1, "Chip Manufacturing LCS"
+# Syntax: -c "set LCS {0}"
+# Syntax: -c "set LCS {1}"
+#
+# CORELIST_S0
+# Specify available physical cores by number
+# Example syntax to connect to physical cores 16 and 17 for S0
+# Syntax: -c "set CORELIST_S0 {16 17}"
+#
+# COREMASK_S0_LO
+# Specify available physical cores 0-63 by mask
+# Example syntax to connect to physical cores 16 and 17 for S0
+# Syntax: -c "set COREMASK_S0_LO {0x0000000000030000}"
+#
+# COREMASK_S0_HI
+# Specify available physical cores 64 and above by mask
+# Example syntax to connect to physical cores 94 and 95 for S0
+# Syntax: -c "set COREMASK_S0_HI {0x00000000C0000000}"
+#
+# PHYS_IDX
+# Enable OpenOCD ARMv8 core target physical indexing
+# If not specified, defaults to OpenOCD ARMv8 core target logical indexing
+# Syntax: -c "set PHYS_IDX {}"
+
+bindto 0.0.0.0
+#debug_level 3
+
+#gdb_port 3333
+#telnet_port 4444
+#tcl_port 6666
+
+#
+# Interface Configuration
+#
+
+source [find interface/jtag_driver.cfg]
+
+#
+# Board (Platform) Configuration
+#
+
+if { ![info exists JTAGFREQ] } {
+	set JTAGFREQ 12375
+}
+
+if { ![info exists SYSNAME] } {
+	set SYSNAME mq
+}
+
+set MQ_ENABLE ""
+source [find board/ampere_qs_mq_1s.cfg]
diff --git a/tcl/openocd.mq_2s_bmc.cfg b/tcl/openocd.mq_2s_bmc.cfg
new file mode 100644
index 000000000..1914d638b
--- /dev/null
+++ b/tcl/openocd.mq_2s_bmc.cfg
@@ -0,0 +1,95 @@
+#
+# Example OpenOCD user configuration for the AST2500 BMC
+# on the Ampere development platforms
+#
+# Copyright (c) 2020-2021, Ampere Computing LLC
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of
+# the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program;
+#
+#
+
+# Command Line Argument Description
+#
+# JTAGFREQ
+# Override the default JTAG clock frequency
+# If not specified, defaults to "12375" kHz
+# Syntax: -c "set JTAGFREQ {12375}"
+#
+# SYSNAME
+# Override the default system name
+# If not specified, defaults to "mq"
+# Syntax: -c "set SYSNAME {mq}"
+#
+# Life-Cycle State (LCS)
+# If not specified, defaults to "Secure LCS"
+# LCS=0, "Secure LCS"
+# LCS=1, "Chip Manufacturing LCS"
+# Syntax: -c "set LCS {0}"
+# Syntax: -c "set LCS {1}"
+#
+# CORELIST_S0, CORELIST_S1
+# Specify available physical cores by number
+# Example syntax to connect to physical cores 16 and 17 for S0 and S1
+# Syntax: -c "set CORELIST_S0 {16 17}"
+# Syntax: -c "set CORELIST_S1 {16 17}"
+#
+# COREMASK_S0_LO, COREMASK_S1_LO
+# Specify available physical cores 0-63 by mask
+# Example syntax to connect to physical cores 16 and 17 for S0 and S1
+# Syntax: -c "set COREMASK_S0_LO {0x0000000000030000}"
+# Syntax: -c "set COREMASK_S1_LO {0x0000000000030000}"
+#
+# COREMASK_S0_HI, COREMASK_S1_HI
+# Specify available physical cores 64 and above by mask
+# Example syntax to connect to physical cores 94 and 95 for S0 and S1
+# Syntax: -c "set COREMASK_S0_HI {0x00000000C0000000}"
+# Syntax: -c "set COREMASK_S1_HI {0x00000000C0000000}"
+#
+# SPLITSMP
+# Group all ARMv8 cores per socket into individual SMP sessions
+# If not specified, group ARMv8 cores from both sockets into one SMP session
+# Syntax: -c "set SPLITSMP {}"
+#
+# PHYS_IDX
+# Enable OpenOCD ARMv8 core target physical indexing
+# If not specified, defaults to OpenOCD ARMv8 core target logical indexing
+# Syntax: -c "set PHYS_IDX {}"
+
+bindto 0.0.0.0
+#debug_level 3
+
+#gdb_port 3333
+#telnet_port 4444
+#tcl_port 6666
+
+#
+# Interface Configuration
+#
+
+source [find interface/jtag_driver.cfg]
+
+#
+# Board (Platform) Configuration
+#
+
+if { ![info exists JTAGFREQ] } {
+	set JTAGFREQ 12375
+}
+
+if { ![info exists SYSNAME] } {
+	set SYSNAME mq
+}
+
+set MQ_ENABLE ""
+source [find board/ampere_qs_mq_2s.cfg]
diff --git a/tcl/openocd.qs_1s_bmc.cfg b/tcl/openocd.qs_1s_bmc.cfg
new file mode 100644
index 000000000..2ee6e517a
--- /dev/null
+++ b/tcl/openocd.qs_1s_bmc.cfg
@@ -0,0 +1,86 @@
+#
+# Example OpenOCD user configuration for the AST2500 BMC
+# on the Ampere development platforms
+#
+# Copyright (c) 2020-2021, Ampere Computing LLC
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of
+# the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program;
+#
+#
+
+# Command Line Argument Description
+#
+# JTAGFREQ
+# Override the default JTAG clock frequency
+# If not specified, defaults to "12375" kHz
+# Syntax: -c "set JTAGFREQ {12375}"
+#
+# SYSNAME
+# Override the default system name
+# If not specified, defaults to "qs"
+# Syntax: -c "set SYSNAME {qs}"
+#
+# Life-Cycle State (LCS)
+# If not specified, defaults to "Secure LCS"
+# LCS=0, "Secure LCS"
+# LCS=1, "Chip Manufacturing LCS"
+# Syntax: -c "set LCS {0}"
+# Syntax: -c "set LCS {1}"
+#
+# CORELIST_S0
+# Specify available physical cores by number
+# Example syntax to connect to physical cores 16 and 17 for S0
+# Syntax: -c "set CORELIST_S0 {16 17}"
+#
+# COREMASK_S0_LO
+# Specify available physical cores 0-63 by mask
+# Example syntax to connect to physical cores 16 and 17 for S0
+# Syntax: -c "set COREMASK_S0_LO {0x0000000000030000}"
+#
+# COREMASK_S0_HI
+# Specify available physical cores 64 and above by mask
+# Example syntax to connect to physical cores 78 and 79 for S0
+# Syntax: -c "set COREMASK_S0_HI {0x000000000000C000}"
+#
+# PHYS_IDX
+# Enable OpenOCD ARMv8 core target physical indexing
+# If not specified, defaults to OpenOCD ARMv8 core target logical indexing
+# Syntax: -c "set PHYS_IDX {}"
+
+bindto 0.0.0.0
+#debug_level 3
+
+#gdb_port 3333
+#telnet_port 4444
+#tcl_port 6666
+
+#
+# Interface Configuration
+#
+
+source [find interface/jtag_driver.cfg]
+
+#
+# Board (Platform) Configuration
+#
+
+if { ![info exists JTAGFREQ] } {
+	set JTAGFREQ 12375
+}
+
+if { ![info exists SYSNAME] } {
+	set SYSNAME qs
+}
+
+source [find board/ampere_qs_mq_1s.cfg]
diff --git a/tcl/openocd.qs_2s_bmc.cfg b/tcl/openocd.qs_2s_bmc.cfg
new file mode 100644
index 000000000..43bea1e51
--- /dev/null
+++ b/tcl/openocd.qs_2s_bmc.cfg
@@ -0,0 +1,94 @@
+#
+# Example OpenOCD user configuration for the AST2500 BMC
+# on the Ampere development platforms
+#
+# Copyright (c) 2020-2021, Ampere Computing LLC
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of
+# the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program;
+#
+#
+
+# Command Line Argument Description
+#
+# JTAGFREQ
+# Override the default JTAG clock frequency
+# If not specified, defaults to "12375" kHz
+# Syntax: -c "set JTAGFREQ {12375}"
+#
+# SYSNAME
+# Override the default system name
+# If not specified, defaults to "qs"
+# Syntax: -c "set SYSNAME {qs}"
+#
+# Life-Cycle State (LCS)
+# If not specified, defaults to "Secure LCS"
+# LCS=0, "Secure LCS"
+# LCS=1, "Chip Manufacturing LCS"
+# Syntax: -c "set LCS {0}"
+# Syntax: -c "set LCS {1}"
+#
+# CORELIST_S0, CORELIST_S1
+# Specify available physical cores by number
+# Example syntax to connect to physical cores 16 and 17 for S0 and S1
+# Syntax: -c "set CORELIST_S0 {16 17}"
+# Syntax: -c "set CORELIST_S1 {16 17}"
+#
+# COREMASK_S0_LO, COREMASK_S1_LO
+# Specify available physical cores 0-63 by mask
+# Example syntax to connect to physical cores 16 and 17 for S0 and S1
+# Syntax: -c "set COREMASK_S0_LO {0x0000000000030000}"
+# Syntax: -c "set COREMASK_S1_LO {0x0000000000030000}"
+#
+# COREMASK_S0_HI, COREMASK_S1_HI
+# Specify available physical cores 64 and above by mask
+# Example syntax to connect to physical cores 78 and 79 for S0 and S1
+# Syntax: -c "set COREMASK_S0_HI {0x000000000000C000}"
+# Syntax: -c "set COREMASK_S1_HI {0x000000000000C000}"
+#
+# SPLITSMP
+# Group all ARMv8 cores per socket into individual SMP sessions
+# If not specified, group ARMv8 cores from both sockets into one SMP session
+# Syntax: -c "set SPLITSMP {}"
+#
+# PHYS_IDX
+# Enable OpenOCD ARMv8 core target physical indexing
+# If not specified, defaults to OpenOCD ARMv8 core target logical indexing
+# Syntax: -c "set PHYS_IDX {}"
+
+bindto 0.0.0.0
+#debug_level 3
+
+#gdb_port 3333
+#telnet_port 4444
+#tcl_port 6666
+
+#
+# Interface Configuration
+#
+
+source [find interface/jtag_driver.cfg]
+
+#
+# Board (Platform) Configuration
+#
+
+if { ![info exists JTAGFREQ] } {
+	set JTAGFREQ 12375
+}
+
+if { ![info exists SYSNAME] } {
+	set SYSNAME qs
+}
+
+source [find board/ampere_qs_mq_2s.cfg]
-- 
2.25.1

